server.js

```diff
@@ -127,18 +132,58 @@ app.use(function (req, res, next) {
 
 /*  Start the Server
 ============================================================================= */
-detectPort(PORT).then(function (_port) {
-  if (PORT !== _port) {
-    console.log('\nPort %d already occupied. Using port %d instead.', PORT, _port)
-    // Persist the new port to the in-memory store. This is kinda hacky
-    // Assign value to cliValue since it overrides all other values
-    var ConfigItem = require('./models/ConfigItem.js')
-    var portConfigItem = ConfigItem.findOneByKey('port')
-    portConfigItem.cliValue = _port
-    portConfigItem.computeEffectiveValue()
-  }
-  http.createServer(app).listen(_port, IP, function () {
-    console.log('\nWelcome to ' + app.locals.title + '!. Visit http://' + (IP === '0.0.0.0' ? 'localhost' : IP) + ':' + _port + BASE_URL + ' to get started')
-  })
-})
+
+var fs = require('fs');
+
+// determine if key pair exists for certs
+if (KEY_PATH.replace(/\s/g,"") != "" && CERT_PATH.replace(/\s/g,"") != "") { //https only
```

Comment:
instead of using replace to clean whitespace, you could do something like ```KEY_PATH.trim()```, which will remove surrounding whitespace and leave any value. Even simpler, js treats an empty string as a falsey, and not empty as truthy, so you could naively do

```js
if (KEY_PATH && CERT_PATH) { 
  // do https setup
}
```

Of course this doesn't work if the user sets one of these values to a white-space padded string, so maybe you'll still want to use .trim()